### ISSUE TITLE
ID overrides when creating models with cancancan

### ISSUE THREAD

**Issue by cdeevfrr** [activescaffold/active_scaffold#617]
### Summary
We are using cancancan and active scaffold, and when we create new models, it tries to use the set of existing IDs as the ID for our new model. 

### Description

We are able to add one a new instance of a model, and that works correctly, but when we try to add the second instance, the database rolls back. In these examples, the second attempt tries to insert with id 60 even though that is the id of the first record.

![screen shot 2018-06-05 at 3 48 07 pm](https://user-images.githubusercontent.com/14035196/40999148-035a80b0-68d8-11e8-8e2d-68248b9b7ca4.png)
*successful creation of first client*

![screen shot 2018-06-05 at 3 48 45 pm](https://user-images.githubusercontent.com/14035196/40999147-034be5aa-68d8-11e8-887a-c68cb4f1ab7d.png)
*resulting record*

![screen shot 2018-06-05 at 3 48 22 pm](https://user-images.githubusercontent.com/14035196/40999146-033f2b94-68d8-11e8-8b6c-defa999bd5a9.png)
*unsuccessful creation attempt of second client*

If we manually add a second instance of the model to the user, the user can then add models like normal and the rollback doesn't occur. 

This has only happened for HABTM models so far, in this example, we have Clients HABTM Websites. 

We have traced this down to the [`new_model`](https://github.com/activescaffold/active_scaffold/blob/0b6067f297d5e060f4a9e35ac4f40c013d9c1064/lib/active_scaffold/actions/core.rb#L272) method in Active scaffold, called in [`do_create`](https://github.com/activescaffold/active_scaffold/blob/0b6067f297d5e060f4a9e35ac4f40c013d9c1064/lib/active_scaffold/actions/create.rb#L93) - it seems that the [`beginning_of_chain`](https://github.com/activescaffold/active_scaffold/blob/bd4a1695633ba634c4722a980bf1e71cc8ce5ec3/lib/active_scaffold/bridges/cancan/cancan_bridge.rb#L45) override in the cancan bridge is returning a list of existing models, and when there is only one model in the list, the id param is being passed all the way back to the new model instance. We verified this by debugginging inside a `before_create_save` hook, and we saw the old IDs being passed to the new models.

This is the relevant line in the ability.rb file, we're using cancancan version 2.1.2 

```
can [:manage], Client, id: client_ids
```

We saw the same bug on revision 378b0c757330292425210889df4c3a3ea3541599 (which we were using originally) and on version 3.5.4 of active scaffold.

Please let me know what other information would be helpful here, or if you think this issue would be better in the cancancan repo!

**Comment by scambra** [comment]
What is client_ids?

**Comment by cdeevfrr** [comment]
@scambra In our `ability.rb` file, we collect a list of clients that a user can manage, and then get the ids for those clients. 

```
client_ids = Client.of_employee(user).pluck(:id)
```

This may be getting into the weeds, but inside `Client.rb` we have:
```
  scope :of_employee, ->(user) { joins(:teams).merge(Team.of_employee(user)) }
```

**Comment by cdeevfrr** [comment]
Our current workaround is to explicitly delete ids before saving when possible:

```
def before_create_save(record)
    record.id = nil
  end
```

However, this bug still causes strange effects for models with custom id columns. In these cases, when a user attempts to create a new instance of the model, the form initially populates the ID column with a list of all the existing IDs that the user has access to.

**Comment by scambra** [comment]
You could clear id in new_model instead of before_create_save, it should fix issue in new and create actions.

```
def new_model
  super.tap { |record| record.id = nil }
end
```

Let me know if that works, then I will add that code to cancan bridge.

**Comment by cdeevfrr** [comment]
This is still on my todo list, I haven't had a chance to give it a try yet...

**Comment by cdeevfrr** [comment]
I gave this a try and it worked. It fixed our problem, and also didn't break one of our models that get their ID as an input from the form, so I think this is a good change!

**Comment by cdeevfrr** [comment]
Please let me know if/when you are able to add the code to cancan bridge.

**Comment by scambra** [comment]
Fixed, both in master and 3.5.5
